fix waddrmgr tests

jolan opened the pull request:
Mostly a conversion of Bitcoin -> Decred addresses and removing some things that don't make sense for Decred.

jcvernaleo commented on wallet/createtx_test.go:
```diff
@@ -0,0 +1,253 @@
+package wallet
```
Where did this file come from?
It is not in dcrwallet master as far as I can tell.  And it doesn't compile.

cjepson commented on waddrmgr/sync.go:
```diff
@@ -16,7 +16,7 @@ const (
 	// for the purposes of rollbacks. Note that if reorganizations longer
 	// than this occur, corruption of the wallet may occur.
-	maxRecentHashes = 1000
+	maxRecentHashes = 20
```
Whoa, that should not be changed. Leave it at 1000.

jolan commented:
There wasn't enough test data for maxRecentHashes to be 1000 which is why I ratcheted it down.  I've added more blocks for it to work and put it back to 1000.

Also removed that errant file that @jcvernaleo mentioned.

jcvernaleo commented:
Looks good now.  Just rebase and squash then OK.

alexlyp commented:
looks good
